hypothesis-python/src/hypothesis/errors.py

```diff
@@ -67,6 +102,19 @@ class Flaky(_Trimmable):
            don't do that and testing those instead.
     """
 
+    def __new__(cls, msg, group):
+        # The Exception mixin forces this an ExceptionGroup (only accepting
+        # Exceptions, not BaseException). Usually BaseException is raised
+        # directly and will hence not be part of a FlakyFailure, but I'm not
+        # sure this assumption holds everywhere. So wrap any BaseExceptions.
```

Comment:
Let's just inherit from `BaseExceptionGroup` to get around that?

Reply:
Unfortunately, the HypothesisException mixing makes it non-base even so. On mobile now, so cant find the ref, but mixing in Exception defines ExceptionGroup.